src/main/java/org/codeforamerica/shiba/pages/SuccessMessageService.java

```diff
@@ -19,80 +21,78 @@ public SuccessMessageService(MessageSource messageSource) {
         }
 
-
-        return String.join("<br><br>", messageKeys);
+        return String.join("<br><br>", paragraphs);
     }
 
-    private List<String> getProgramNames(Locale locale, List<String> applicantPrograms, boolean hasNonExpeditedSnap, boolean isCcapExpeditedEligible) {
-        boolean hasCcap = applicantPrograms.stream().anyMatch(p -> p.equals(Program.CCAP));
-        boolean hasNonExpeditedCcap = hasCcap && !isCcapExpeditedEligible;
-        boolean hasGrh = applicantPrograms.stream().anyMatch(p -> p.equals(Program.GRH));
-        boolean hasCash = applicantPrograms.stream().anyMatch(p -> p.equals(Program.CASH));
-        boolean hasEa = applicantPrograms.stream().anyMatch(p -> p.equals(Program.EA));
-        List<String> nextStepLetterProgramNames = new ArrayList<>();
+    private List<String> getNextStepLetterPrograms(List<String> allPrograms,
+                                                   boolean isSnapExpeditedEligible,
+                                                   boolean isCcapExpeditedEligible,
+                                                   LocaleSpecificMessageSource ms) {
+        boolean hasNonExpeditedSnap = hasProgram(SNAP, allPrograms) && !isSnapExpeditedEligible;
+        boolean hasNonExpeditedCcap = hasProgram(CCAP, allPrograms) && !isCcapExpeditedEligible;
+
+        List<String> nextStepLetterPrograms = new ArrayList<>();
         if (hasNonExpeditedCcap) {
-            nextStepLetterProgramNames.add(getMessage("success.childcare", locale));
+            nextStepLetterPrograms.add(ms.getMessage("success.childcare"));
         }
-        if (hasGrh) {
-            nextStepLetterProgramNames.add(getMessage("success.housing", locale));
+        if (hasProgram(GRH, allPrograms)) {
+            nextStepLetterPrograms.add(ms.getMessage("success.housing"));
         }
-        if (hasEa) {
-            nextStepLetterProgramNames.add((getMessage("success.emergency-assistance", locale)));
+        if (hasProgram(EA, allPrograms)) {
+            nextStepLetterPrograms.add((ms.getMessage("success.emergency-assistance")));
         }
-        if (hasCash) {
-            nextStepLetterProgramNames.add((getMessage("success.cash-support", locale)));
+        if (hasProgram(CASH, allPrograms)) {
+            nextStepLetterPrograms.add((ms.getMessage("success.cash-support")));
         }
         if (hasNonExpeditedSnap) {
-            nextStepLetterProgramNames.add(getMessage("success.food-support", locale));
+            nextStepLetterPrograms.add(ms.getMessage("success.food-support"));
         }
 
-        return nextStepLetterProgramNames;
+        return nextStepLetterPrograms;
     }
 
-    @NotNull
-    private String getMessage(String messageKey, Locale locale) {
-        return messageSource.getMessage(messageKey, null, locale);
+    private boolean hasProgram(String program, List<String> programs) {
+        return programs.stream().anyMatch(p -> p.equals(program));
```

Comment:
Could this just be a `programs.contains(program)`?

Reply:
YES great point, thank you! I'll change that